server/app/models/host_node.rb

```diff
@@ -45,9 +47,10 @@ class HostNode
   after_save :reserve_node_number, :ensure_unique_name
 
   index({ grid_id: 1 })
-  index({ node_id: 1 })
+  index({ node_id: 1 }, { unique: true, sparse: true })
```

Comment:
Added. Need to just drop the old index and re-create with the new options, or it fails:

```
I, [2017-06-30T07:38:48.892181 #243]  INFO -- Mongodb::Migrator: migrating HostNodeIndexes exception: Index with name: node_id_1 already exists with different options (85)
```